outlines/models/utils.py

```diff
@@ -0,0 +1,58 @@
+def set_additional_properties_false_json_schema(schema: dict) -> dict:
```

Comment:
Would using `jsonpath` simplify things here?

Reply:
Ah I did not know about it, much simpler indeed!